[Teo Zi Hui, Shelia] iP

shelialia opened the pull request:
# Task Assistant Rose
> You don't have to see the whole staircase, just take the first step" — Martin Luther King, Jr. [View Source](https://www.goodreads.com/quotes/1063592-you-don-t-have-to-see-the-whole-staircase-just-take)

Here's how Rose makes _your_ day better
* Keeps track of your basic tasks, deadlines, and events
* Easy search up for tasks
* Easy addition and deletion
* and more upcoming!! 💯 

But what's so special about Rose?
1. Run here directly from your laptop
2. Get rid of websites, just run her locally
3. Your personal assistant

And it is *FREE*!

I used Java to build the app. Here is the main entry point of the application.
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
Why not you try it for yourself?

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

shelialia committed:
Add .gitignore to exclude unnecessary files

shelialia committed:
Level 0: Rename, Greet, Exit

shelialia committed:
Level 1: Echo

shelialia committed:
Level 2: Add and List

shelialia committed:
Level 3: Mark as Done

shelialia committed:
Level 3: Wrote Todo class

shelialia committed:
Level 3: Wrote Deadline class

shelialia committed:
Level 3: Wrote Event class

shelialia committed:
Level 3: Edited chatbot logic to use written classes

shelialia committed:
corrected letter spacing in string output

shelialia committed:
corrected letter spacing in string output again

shelialia committed:
another letter spacing correction for string output

shelialia committed:
A-TextUiTesting: wrote actual and expected test case

shelialia committed:
Level-5: Handle Errors

shelialia committed:
Level 6: changed string array into arraylist

shelialia committed:
Level 6: Delete Tasks

shelialia committed:
A-Enums: Use Enums

shelialia committed:
Implement Storage class for saving and loading tasks. Update Task classes to support file formatting. Integrate Storage with Rose chatbot

shelialia committed:
Merge Level-7: Save functionality

shelialia committed:
Add date time format feature to deadline class

shelialia committed:
Merge Level-8: Save functionality

shelialia committed:
added TaskList, Ui, Parser class and refactored Rose code to work with them

shelialia committed:
change LocalDate to LocalDateTime object in Deadline class

shelialia committed:
fix bugs related to printing

shelialia committed:
A-MoreOOP: Use More OOP

shelialia committed:
Organize the classes into rose.ui

shelialia committed:
Merge branch 'A-Packages'

shelialia committed:
fix merge conflicts

shelialia committed:
Merge branch 'refs/heads/A-Gradle'

shelialia committed:
added junit test cases

shelialia committed:
add JavaDoc comments for all classes

shelialia committed:
tweak code to comply with coding standard

shelialia committed:
add findTasks function in TaskList to simulate searching for tasks among tasklist

shelialia committed:
add feature for user to find task by searching for a keyword in the task description

shelialia committed:
Merge branch 'branch-A-JavaDoc'
 the commit.

shelialia committed:
fixed merge conflicts

shelialia committed:
resolve merge conflicts

tim0tay commented on src/main/java/rose/ui/Event.java:
```diff
@@ -0,0 +1,52 @@
+     * @return A string representing this event in file format.
+     */
+    @Override
+    public String toFileFormat() {
```
I like that the naming of this method (and for the other tasks too) are clearly understandable!

tim0tay commented on src/main/java/rose/ui/Rose.java:
```diff
@@ -0,0 +1,126 @@
+            try {
+                Command command = Parser.parse(input);
+
+                switch (command.getCommand().toLowerCase()) {
```
Would it be better to encapsulate this parsing in the parser class?

tim0tay commented on src/main/java/rose/ui/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+package rose.ui;
```
Perhaps these tasks (Deadline, Event, Todo) can be in a Task package instead?

tim0tay commented on src/main/java/rose/ui/Storage.java:
```diff
@@ -0,0 +1,97 @@
+        boolean isDone = parts[1].equals("1");
+
+        switch (taskType) {
+            case "T":
```
Mayhaps 'tis better to not indent cases!

tim0tay commented on src/main/java/rose/ui/Ui.java:
```diff
@@ -0,0 +1,90 @@
+     * Displays the greeting message when the program starts.
+     */
+    public void greet() {
+        System.out.println("____________________________________________________________");
```
I love how you separated the horizontal line into two lines! 👍

tim0tay reviewed (commented):
Shelia this looks amazing! Can tell that you've put in a lot of effort and diligence into ensuring that your code is of good quality :> 🥇

shelialia committed:
created basic gui and user interactions

shelialia committed:
finish part 4 of javafx tutorial

shelialia committed:
set the dimensions using scenebuilder

shelialia committed:
set up and link css to fxml. flip text bubbles

shelialia committed:
add background image

shelialia committed:
changed daduke's photo

shelialia committed:
edit Rose.java logic to send responses as a string to the gui

shelialia committed:
Add assertions to Storage, Tasks, and Ui classes

Assertions have been added to enforce key assumptions in the Storage, Tasks, and Ui classes, improving code robustness and preventing unintended behavior.

- Storage: Ensures file paths to load from and the line from the file representing a task is not null.
- Tasks: Validates that only non-empty keywords can be used to search for a task.
- Ui: Prevents null strings from being returned and displayed.

These assertions help detect potential issues early during development while remaining disabled in production to avoid performance overhead.

shelialia committed:
Add case of finding for task via keyword

shelialia committed:
Add assert to Parser class in parse user input method

shelialia committed:
Storage: add assertions to ensure data integrity

Currently, the Storage class lacks assertions to validate critical conditions,
which could lead to unexpected null values or corrupted data being processed
without early detection.

Assertions help catch logical errors during development by enforcing constraints
on input values and ensuring expected conditions hold.

Let's,
* add an assertion to ensure  is not null or empty in the constructor
* add an assertion to check that parsed task lines are not null or empty

shelialia committed:
Remove unused code in Rose file

shelialia committed:
Merge pull request #1 from shelialia/branch-A-Assertions

Branch a assertions

shelialia committed:
Fix merge conflicts

shelialia committed:
Remove code calling run() method

shelialia committed:
Resolve merge conflicts

shelialia committed:
Merge branch 'master' into branch-A-CodeQuality

shelialia committed:
Merge pull request #2 from shelialia/branch-A-CodeQuality

Add case of finding for task via keyword

shelialia committed:
Merge remote-tracking branch 'origin/master'

shelialia committed:
Add gradle.yml file to set up CI

shelialia committed:
Add functionality to update Event start and end time

shelialia committed:
Fix DeadlineTest testcase expected output

shelialia committed:
Add JavaDoc for command package

shelialia committed:
Add JavaDoc for gui package

shelialia committed:
Add JavaDoc for parser package

shelialia committed:
Add JavaDoc for storage package

shelialia committed:
Add JavaDoc for tasks package

shelialia committed:
Fix bug causing invocation error when unknown command sent

shelialia committed:
Ammend name of image files, discarding default Duke name

shelialia committed:
Rename all traces of Rose to PerryGPT

shelialia committed:
Add readme user guide

shelialia committed:
Add datetime formatter for event task

shelialia committed:
Update README to for event task format

shelialia committed:
Add title to GUI

shelialia committed:
Update README.md with product snapshot

shelialia committed:
Add launcher and refactor build.gradle to point to correct launch path

shelialia committed:
Update README.md

shelialia committed:
Fix help command when wrong command given

shelialia committed:
Fix error in README.md regarding example deadline command

shelialia committed:
Rename img.png to Ui.png

shelialia committed:
Add try catch block to Deadline to throw exception when given incorrect date format

shelialia committed:
Throw exception when given incorrect date format for Deadline and Event

shelialia committed:
Add unmark command under command package, and add case under parser class

shelialia committed:
Correct minor typo in README.md

shelialia committed:
Edit JavaDoc for Storage

shelialia committed:
Edit JavaDoc for Task

shelialia committed:
Edit JavaDoc for UnmarkCommand

shelialia committed:
Edit README.md introduction

shelialia committed:
Edit JavaDoc for ByeCommand

shelialia committed:
Update README.md to add more details on the mark and unmark tasks section

shelialia committed:
Update README.md to standardize for find all tasks section

shelialia committed:
Update README.md to standardize for delete tasks by task ID section

shelialia committed:
Update README.md to standardize for mark or unmark tasks by task ID section

shelialia committed:
Update README.md to standardize for ending conversation with Ferb

shelialia committed:
Update README.md

shelialia committed:
Update README.md

shelialia committed:
Update README.md

shelialia committed:
Update README.md

shelialia committed:
Update README.md
